resolving/npm-resolver/src/parsePref.ts

```diff
@@ -36,16 +34,53 @@ export function parsePref (
       }
     }
   }
-  if (pref.startsWith(registry)) {
-    const pkg = parseNpmTarballUrl(pref)
+  if (tag.startsWith(registry)) {
+    const pkg = parseNpmTarballUrl(tag)
     if (pkg != null) {
       return {
         fetchSpec: pkg.version,
         name: pkg.name,
-        normalizedPref: pref,
+        normalizedPref: tag,
         type: 'version',
       }
     }
   }
   return null
 }
+
+export function parsePref (
+  pref: string,
+  alias: string | undefined,
+  defaultTag: string,
+  registry: string
+): RegistryPackageSpec | null {
+  let name = alias
+  if (pref.startsWith('npm:')) {
+    [name, pref] = parseNameAndTag(pref.slice('npm:'.length), defaultTag)
+  }
+  return parsePrefFromNameAndTag(name, pref, registry)
+}
+
+export function parseJsrPref (
+  pref: string,
+  alias: string | undefined,
+  defaultTag: string,
+  registry: string
+): RegistryPackageSpec | null {
+  if (!pref.startsWith('jsr:')) return null
+  pref = pref.slice('jsr:'.length)
+  let spec = parsePref(pref, alias, defaultTag, registry)
+  if (!spec) {
+    const [name, tag] = parseNameAndTag(pref, defaultTag)
+    spec = parsePrefFromNameAndTag(name, tag, registry)
+  }
+  if (!spec) {
+    throw new PnpmError('INVALID_JSR_SPECIFICATION', `Cannot parse '${pref}' as an npm specification`)
+  }
+  if (!spec.name.startsWith('@')) {
+    throw new PnpmError('MISSING_JSR_PACKAGE_SCOPE', 'Package names from JSR must have scopes')
+  }
+  const jsrNameSuffix = spec.name.replace('@', '').replace('/', '__') // not replaceAll because we only replace the first of each character
+  spec.name = `@jsr/${jsrNameSuffix}`
+  return spec
+}
```

Comment:
I wanted to reuse `@pnpm/jsr-specs` in this function, but I don't fully understand what `parsePref` does.

Reply:
https://github.com/pnpm/pnpm/pull/9358/commits/ed2ed0b67167bce1f5682319706c3d0f33c589d3